pip_install_process.go

```diff
@@ -0,0 +1,45 @@
+package pip
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/paketo-buildpacks/packit/pexec"
+)
+
+//go:generate faux --interface Executable --output fakes/executable.go
+
+// Executable defines the interface for invoking an executable.
+type Executable interface {
+	Execute(pexec.Execution) error
+}
+
+// PipInstallProcess implements the InstallProcess interface.
+type PipInstallProcess struct {
+	executable Executable
+}
+
+// NewPipInstallProcess creates an instance of the PipInstallProcess given an Executable that runs `python`.
+func NewPipInstallProcess(executable Executable) PipInstallProcess {
+	return PipInstallProcess{
+		executable: executable,
+	}
+}
+
+// Execute installs the pip binary from source code located in the given srcPath into the a layer path designated by targetLayerPath.
+func (p PipInstallProcess) Execute(srcPath, targetLayerPath string) error {
+	buffer := bytes.NewBuffer(nil)
+
+	err := p.executable.Execute(pexec.Execution{
+		Args:   []string{"-m", "pip", "install", srcPath, "--user"},
```

Comment:
I understand we are not dealing with pip cache dir at this point, but is there a reason this only installed with `--user` and not other flags like we used to do here: https://github.com/paketo-community/pip/blob/main/pip/pip.go#L54-L60?
Documentation says something like `--compile` can produce bytecode which I assume is faster. I haven't done much research but like to hear your thoughts

Reply:
Personally, I didn't research the benefits of using `--compile`. The goal here is just to install pip, and not worry about anything in the `requirements.txt`, so I'm not sure if it would be overkill to run the command with a load of flags other flags. 
I think we found the simplest thing that works for our use case of just installing pip, but there's definitely a good chance that some of the other flags could be useful.

Reply:
It feels like options like `--upgrade-strategy only-if-needed --ignore-installed` could be useful if python came with the version we are trying to install.
@thitch97 @ryanmoran Do you have any context/insight into the options used in current main?